week-2-wide-perspective/w02e04-dungeons-and-the-princess/src/model/castle.ts

```diff
@@ -0,0 +1,25 @@
+import { Dungeon } from './dungeon';
+import { Person } from './person';
+
+export class Castle {
+    constructor(private dungeon: Dungeon) { }
+
+    gotoTheDungeon(daredevil: Person) {
+        this.throwIfPersonNotAllow(daredevil);
+        this.dungeon.enter(daredevil)
+    }
+
+    private throwIfPersonNotAllow(person: Person): void {
```

Comment:
To jest sprytne, ale żeby identyfikować "skąd" są błędy - a nie tylko kto je wywołuje - to nie może być tak zgeneralizowane.

Każdy błąd powinien być rzucany z konkretnego miejsca. Np. 
`Queen` nie może być w `Dungeon` i to właśnie stamtąd musisz rzucić błąd + wykombinować skąd on zostął rzucony własnie na górze np. w `main.ts`

- choć w Twoim przypadku @Dam1anM będzie to w testach.